nix/packages/grub-efi-image.nix

```
@@ -0,0 +1,47 @@
+{ lib, stdenvNoCC, grub2_efi, fetchFromGitHub, git, gnulib }:
+let
+  grub_uki = grub2_efi.overrideAttrs (finalAttrs: previousAttrs: rec {
+    patches = [ ];
+    src = fetchFromGitHub {
+      owner = "GovanifY";
+      repo = "grub";
+      rev = "master";
+      sha256 = "sha256-2XhZdRCK73QJ5QIeTdqXVFinzXq36pJ/6EwnluY5rnA=";
+    };
+
+
+    prePatch = ''
+      patchShebangs autogen.sh bootstrap
```

```suggestion
    postPatch = ''
      patchShebangs autogen.sh bootstrap
```
You don't want to change files in prePatch because then patches from patches could no longer apply.

---

This needs to execute _before_ the autoreconf phase, which is executed before configure and I've emptied the patches list in case you didn't see right after this snippet of code. So unless you have another idea on how to do this cleanly without rewriting the entire derivation I'll keep this one as is I think.

---

Sorry, I wasn't super clear about this. The patching part should be in postPatch and there should be a preAutoreconfPhase

---

Ahhh, misunderstood. Done!